deeprankcore/dataset.py

```diff
@@ -205,6 +209,133 @@ def len(self) -> int:
+            If bins is a string, it is one of the binning strategies supported by numpy.histogram_bin_edges:
+            'auto', 'fd', 'doane', 'scott', 'stone', 'rice', 'sturges', or 'sqrt'.
+            Defaults to 10.
+        
+        figsize (tuple): saved figure sizes, defaults to (15, 15).
+        """
+        if self.df is None:
+            raise ValueError("Please first read in the data into a pd.DataFrame using `hdf5_to_pandas` method of the class.")
+        
+        if not isinstance(features, list):
+            features = [features]
+
+        features_df = [col for feat in features for col in self.df.columns.values.tolist() if feat in col]
+
+        means = [
+            round(np.concatenate(self.df[feat].values).mean(), 1) if isinstance(self.df[feat].values[0], np.ndarray) \
+            else round(self.df[feat].values.mean(), 1) \
+            for feat in features_df]
+        devs = [
+            round(np.concatenate(self.df[feat].values).std(), 1) if isinstance(self.df[feat].values[0], np.ndarray) \
+            else round(self.df[feat].values.std(), 1) \
+            for feat in features_df]
+
+        if len(features_df) > 1:
+
+            fig, axs = plt.subplots(len(features_df), figsize=figsize)
+
+            for row, feat in enumerate(features_df):
+
+                if isinstance(self.df[feat].values[0], np.ndarray):
+                    axs[row].hist(np.concatenate(self.df[feat].values), bins=bins)
+                else:
+                    axs[row].hist(self.df[feat].values, bins=bins)
+                axs[row].set(xlabel=f'{feat} (mean {means[row]}, std {devs[row]})', ylabel='Count')
+            fig.tight_layout()
+
+        elif len(features_df) == 1:
+
+            fig = plt.figure(figsize=figsize)
+            ax = fig.add_subplot(111)
+            if isinstance(self.df[features_df[0]].values[0], np.ndarray):
+                ax.hist(np.concatenate(self.df[features_df[0]].values), bins=bins)
+            else:
+                ax.hist(self.df[features_df[0]].values, bins=bins)
+            ax.set(xlabel=f'{features_df[0]} (mean {means[0]}, std {devs[0]})', ylabel='Count')
+
+        else:
+            raise ValueError("Please provide valid features names. They must be present in the current :class:`DeeprankDataset` children instance.")
+        
+        fig.tight_layout()
+        fig.savefig(fname)
+        plt.close(fig)
+    
+    def _compute_mean_std(self):
+
+        means = {col: round(np.concatenate(self.df[col].values).mean(), 1) if isinstance(self.df[col].values[0], np.ndarray) \
+            else round(self.df[col].values.mean(), 1) \
+            for col in self.df.columns[1:]}
+        devs = {col: round(np.concatenate(self.df[col].values).std(), 1) if isinstance(self.df[col].values[0], np.ndarray) \
+            else round(self.df[col].values.std(), 1) \
+            for col in self.df.columns[1:]}
+        self.means = means
+        self.devs = devs
```

Comment:
It requires more than a quick fix since in `_compute_mean_std` means and devs are assigned to the attributes and this may mess up the train/test/valid logic in which means and devs refer to the training values, and not to the valid or testing one. I would say it would be better to have a function that outputs means and dev and then outside of it we do attributes assignation, but since the operation of computing means and devs is not a bottleneck (computing the df is) I'm not sure it's worth it now. Maybe we can open a separate issue about it?

Reply:
Agreed.
Can you create a nice-to-have issue for this?